src/cpp/fastdds/publisher/DataWriterHistory.cpp

```diff
@@ -522,6 +538,29 @@ bool DataWriterHistory::wait_for_acknowledgement_last_change(
     return false;
 }
 
+bool DataWriterHistory::change_is_acked_or_fully_delivered(
+        const CacheChange_t* change)
+{
+    bool is_acked = false;
+    if (mp_writer->getAttributes().reliabilityKind == ReliabilityKind_t::RELIABLE)
+    {
+        auto stateful_writer = dynamic_cast<StatefulWriter*>(mp_writer);
```

Comment:
I don't like this cast. If we pass a `bool has_disable_positive_acks` to the constructor, we wouldn't need to query the writer for it. Another option is making `get_disable_positive_acks()` a virtual method on `RTPSWriter`.

Reply:
The second option has been chosen. As this implies an ABI break, it has been added to the `versions.md` file.

Reply:
Done in 95f8b55d5 and e89d49ffb